apps/oxlint/src-js/plugins/scope.ts

```diff
@@ -124,7 +229,42 @@ export function getScope(node: Node): Scope {
  * @param refNode? - The closest node to the variable reference.
  * @returns `true` if the variable was found and marked as used, `false` if not.
  */
-// oxlint-disable-next-line no-unused-vars
-export function markVariableAsUsed(name: string, refNode: Node): boolean {
-  throw new Error('`sourceCode.markVariableAsUsed` not implemented yet'); // TODO
+export function markVariableAsUsed(name: string, refNode?: ESTree.Node): boolean {
```

Comment:
Can we explicitly remove support for this?

The reason this exists, is because it allows plugins e.g. jsdoc to mark a variable as used so that no-unused-vars doesn't support this.

We have a problem in that no-unused-vars runs on the rust side, before the JS rules run, so this won't really work as expected